src/main/java/org/spongepowered/common/data/processor/data/entity/PassengerDataProcessor.java

```diff
@@ -0,0 +1,177 @@
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<PassengerData> fill(DataHolder dataHolder, PassengerData manipulator, MergeFunction overlap) {
+        if (dataHolder instanceof EntityLivingBase) {
+            final PassengerData merged = overlap.merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+            manipulator.set(Keys.BASE_VEHICLE, merged.baseVehicle().get())
+                    .set(Keys.VEHICLE, merged.vehicle().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<PassengerData> fill(DataContainer container, PassengerData passengerData) {
+        passengerData.set(Keys.BASE_VEHICLE, getData(container, Keys.BASE_VEHICLE));
+        passengerData.set(Keys.VEHICLE, getData(container, Keys.VEHICLE));
+        return Optional.of(passengerData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, PassengerData manipulator) {
+        if (dataHolder instanceof EntityLivingBase) {
+            DataTransactionBuilder builder = DataTransactionBuilder.builder();
+            final Entity prevVehicle = (Entity) ((EntityLivingBase) dataHolder).riddenByEntity;
+            final Entity prevBaseVehicle = (Entity) ((EntityLivingBase) dataHolder).riddenByEntity;
+            final EntityLivingBase livingBase = ((EntityLivingBase) dataHolder);
+            final Entity newBaseVehicle = manipulator.baseVehicle().get();
+            final Entity newVehicle = manipulator.vehicle().get();
+
+            try {
+                builder.replace(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, prevVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, prevBaseVehicle));
+                livingBase.ridingEntity = (net.minecraft.entity.Entity) newVehicle;
+                builder.success(
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle))
+                        .result(DataTransactionResult.Type.SUCCESS);
+                return builder.build();
+            } catch (Exception e) {
+                livingBase.riddenByEntity = (net.minecraft.entity.Entity) prevVehicle;
+                builder.reject(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle))
+                        .result(DataTransactionResult.Type.ERROR);
+                return builder.build();
+            }
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, PassengerData manipulator, MergeFunction function) {
+        return null;
```

Comment:
Not sure if it's needed but a //TODO: finish Data transaction stuff would make sure this isn't missed.